stactask/task.py

```diff
@@ -108,15 +107,29 @@ def __init__(
 
     @property
     def process_definition(self) -> dict[str, Any]:
-        process = self._payload.get("process", {})
+        process = self._payload.get("process", [])
         if isinstance(process, dict):
+            warnings.warn(
+                "`process` is dictionary, use a list instead",
+                DeprecationWarning,
+                stacklevel=2,
+            )
             return process
-        else:
-            raise ValueError(f"process is not a dict: {type(process)}")
+
+        if not isinstance(process, list):
+            raise TypeError("`process` must be a list of dictionaries")
+
+        if not process:
+            return {}
+
+        if not isinstance(process[0], dict):
+            raise TypeError("`process` list elements must be dictionaries")
+
+        return process[0]
```

Comment:
@jkeifer I'm following in [`cirrus-geo`'s footsteps](https://github.com/cirrus-geo/cirrus-geo/blob/a909b54c123f166d34af05db34de74f6a10d74d4/src/cirrus/lib/process_payload.py#L56) here. I'll need some context/guidance on how to deal with the list of process blocks, i.e., how do we either iterate through the list of process blocks or otherwise select the appropriate process block from the supplied list?

Reply:
It's always the first process definition in the list, as you have.